### ISSUE TITLE
Missing Vim features

### ISSUE THREAD

**Issue by justin8** [ajaxorg/ace#2680]
Just putting in a check list of features I've noticed missing. Please add more as they are found:
- [ ] `%s` (for cloud9)
- [ ] `=`
- [x] `:sort`
- [ ] `:vsplit` splits the pane, but uses an existing document; doesn't allow you to vsplit the same file in multiple windows
- [x] "tag" motion commands: `cit` = change inner tag, `dat` = delete a tag

**Comment by xEtherealx** [comment]
+1 this is pretty important, esp. find/replace

**Comment by 22a** [comment]
`%s` seems to work as expected for me. Do these tests neglect the functionality you desire?
https://github.com/ajaxorg/ace/blob/master/lib/ace/keyboard/vim_test.js#L3529-L3767

**Comment by nightwing** [comment]
@22a %s doesn't work in cloud9.

**Comment by fam4r** [comment]
Do clipboard keybindings share content with the system clipboard?

**Comment by nightwing** [comment]
Currently only ctrl-c does, which other keybindings should share content with the system clipboard?

**Comment by fam4r** [comment]
It would be nice to have Vim copy/cut-paste keybindings sharing the system clipboard

**Comment by fam4r** [comment]
Keep it open

**Comment by whazor** [comment]
It looks like the vim mode has `navigator.clipboard.readText` in the code, but it was not triggered, and it was broken. I found root causes and created PR (#5073). Though the tests seem to fail for now as our tester does not have `navigator.query`.

**Comment by gtritchie** [comment]
Add `:set list` to the wishlist. Here's the description from the vim docs:

### VIEWING TABS

When there are tabs in a file, you cannot see where they are.  To make them
visible: 

	:set list

Now every tab is displayed as ^I.  And a $ is displayed at the end of each
line, so that you can spot trailing spaces that would otherwise go unnoticed.

**Comment by nightwing** [comment]
@gtritchie should this simply forward to the existing `showInvisibles` option of ace, or are you looking for an implementation more closely imitating vim behavior (e.g. including `listchars`)?

```js
require("ace/keyboard/vim").Vim.defineOption("list", false, 'boolean', null, function(value, cm) {
    if (cm === undefined) {
        return;
    }
    if (value === undefined) {
        var value = cm.ace.getOption('showInvisibles');
        return value;
    }
    else {
        cm.ace.setOption('showInvisibles', value);
    }
})
```

I guess the tricky bit here is finding a way to make this work well with the changes embedding app makes using the api

**Comment by gtritchie** [comment]
@nightwing Good point. Our integration already exposes Ace's `showInvisibles` via a user option.

I was thinking of an imitation of vim's default behavior (w/o worrying about listchars), but perhaps a good-enough approach would be to toggle `showInvisibles` with `set list` and `set nolist` and call it a day (which I believe we can do on our end fairly easily).

**Comment by Leonardo-Boss** [comment]
Please add `:norm` and `:{range}norm` to the wishlist. Here's the description from the docs.
# :norm
			Execute Normal mode commands {commands}.  This makes
			it possible to execute Normal mode commands typed on
			the command-line.  {commands} are executed like they
			are typed.  For undo all commands are undone together.
			Execution stops when an error is encountered.

# :{range}norm
			Execute Normal mode commands {commands} for each line
			in the {range}.  Before executing the {commands}, the
			cursor is positioned in the first column of the range,
			for each line.  Otherwise it's the same as the
			":normal" command without a range.

**Comment by nightwing** [comment]
@Leonardo-Boss  `:norm` and several other missing commands are already implemented on master https://raw.githack.com/ajaxorg/ace/master/kitchen-sink.html, and will be released soon.
